src/measure_theory/covering/liminf_limsup.lean

```diff
@@ -0,0 +1,196 @@
+/-
+Copyright (c) 2022 Oliver Nash. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Oliver Nash
+-/
+import measure_theory.covering.density_theorem
+
+/-!
+# Liminf, limsup, and doubling measures.
+
+This file is a place to collect lemmas about liminf and limsup for subsets of a metric space
+carrying a doubling measure.
+
+## Main results:
+
+ * `blimsup_cthickening_mul_ae_eq`: the limsup of the closed thickening of a sequence of compact
+   subsets is unchanged if the sequence of distances is multiplied by a positive scale factor.
```

Comment:
With this phrasing, it looks like a trivial formal statement. Which it is not, because it is only unchanged almost everywhere. This is maybe worth expanding a little bit.

Reply:
Good point, done.